django/db/migrations/serializer.py

```diff
@@ -270,62 +271,64 @@ def serialize(self):
         return "uuid.%s" % repr(self.value), {"import uuid"}
 
 
+def is_deconstructs(value):
+    unsuitable_types = (models.Field, models.manager.BaseManager, Operation, type)
+    return not isinstance(value, unsuitable_types) and hasattr(value, 'deconstruct')
+
+
+class RegisterSerialize:
+    @classmethod
+    def register_lookup(cls, type_, serializer):
+        if not issubclass(serializer, BaseSerializer):
+            raise ValueError("'%s' must inherit from 'BaseSerializer'. ", serializer)
+        if 'class_lookups' not in cls.__dict__:
+            cls.get_registry()
+        cls.class_lookups[type_] = serializer
+        return type_
+
+    @classmethod
+    def get_registry(cls):
+        if 'class_lookups' in cls.__dict__:
```

Comment:
Did you consider using `cached_property` rather than implementing your own caching?